cli/main.cpp

```diff
@@ -61,7 +61,9 @@
 
 #include "cppcheckexecutor.h"
 
+#ifdef NDEBUG
```

Comment:
this feels overzealous to me .. if we need to have iostream anyway it does not hurt to include it. the code becomes more cludgy here not less.

Reply:
I actually remover that since I was not using NDEBUG so it would be helpful for later runs to know that I may not remove this. I find this cleaned than adding some kind of random or `IWYU: keep` comment to it.

The existing defines for some includes are way worse as they should not be using platforms but more generic checks - but I will not dare touching them.

Reply:
I don't want to have such comments neither. they don't improve readability. This ifdef does not improve the readability neither.

You should ideally have fixed iwyu instead. This include iostream is required and you can't remove it.

Reply:
iwyu is correct - the include is not needed unless `NDEBUG` is defined.

Reply:
yes I understand that.

Imho the point of iwyu is to cleanup the code.